packages/status-js/src/utils/encode-url-data.test.ts

```diff
@@ -0,0 +1,92 @@
+import { describe, expect, test } from 'vitest'
+
+import {
+  decodeChannelUrlData,
+  decodeCommunityUrlData,
+  decodeUserUrlData,
+  encodeChannelUrlData,
+  encodeCommunityUrlData,
+  encodeUserUrlData,
+} from './encode-url-data'
+
+describe('URL data', () => {
+  test('should encode and decode community', () => {
+    const data = {
+      displayName: 'Lorem ipsum dolor sit egestas.',
+      description:
+        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus non dui vitae augue elementum laoreet ac pharetra odio. Morbi vestibulum.',
+      membersCount: 1_000_000,
+      color: '#4360DF',
+    }
+    const publicKey = 'zQ3shY7r4cAdg4eUF5dfcuCqCFzWmdjHW4SX5hspM9ucAarfU'
+
+    const encodedData = encodeCommunityUrlData(data, publicKey)
+    const decodedData = decodeCommunityUrlData(encodedData, publicKey)
+
+    expect(encodedData).toBe(
+      'G8QAgB2HsXukxPjOtKWIubcNl-B0uURsnKDo9aJV-0VzRbRGSUCoqm5G0xr2x3HvncEPqJXanmOcKGy6yMLbtw8espSvaBJj6QDp8OCCzq2V8uEqwVGdSfACIGwxf7i4WCazAtZ1dF_tAU5lm3De56yL7Ru9hE9phNUowsGzFqZ5kAl7cf4WzDB4lzebHo_w8aaempKoyas6hpxKbJ3d_wY='
+    )
+    expect(decodedData).toEqual(data)
+  })
+
+  test('should throw on invalid checksum for community encoded with incorrect public key', () => {
+    expect(() => {
+      const originalData = {
+        displayName: 'Lorem ipsum dolor sit egestas.',
+        description:
+          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus non dui vitae augue elementum laoreet ac pharetra odio. Morbi vestibulum.',
+        membersCount: 1_000_000,
+        color: '#4360DF',
+      }
+      const myPublicKey = 'zQ3shY7r4cAdg4eUF5dfcuCqCFzWmdjHW4SX5hspM9ucAarfU'
+      const notMypublicKey = 'zQ3shQkUHszVrsLxQ8ecNHfKg1P2W8EUnmZC7EWUVy6NbL6pa'
+
+      const encodedData = encodeCommunityUrlData(originalData, notMypublicKey)
+      decodeCommunityUrlData(encodedData, myPublicKey)
+    }).toThrowErrorMatchingInlineSnapshot(`"Invalid checksum"`)
+  })
+
+  test('should encode and decode  channel', () => {
```

Comment:
nitpick: it looks like there are two spaces between "decode" and "channel"

Reply:
```suggestion
  test('should encode and decode channel', () => {
```